packages/jumpstarter-driver-snmp/jumpstarter_driver_snmp/driver.py

```diff
@@ -0,0 +1,166 @@
+            # TODO: switch to anyio?
+            asyncio.get_running_loop()
+        except RuntimeError:
+            loop = asyncio.new_event_loop()
+            asyncio.set_event_loop(loop)
+
+
+        snmp_engine = engine.SnmpEngine()
+
+        config.add_v3_user(
+            snmp_engine,
+            self.user,
+            config.USM_AUTH_NONE,
+            None
+        )
+        config.add_target_parameters(
+            snmp_engine,
+            "my-creds",
+            self.user,
+            "noAuthNoPriv"
+        )
+
+        config.add_transport(
+            snmp_engine,
+            udp.DOMAIN_NAME,
+            udp.UdpAsyncioTransport().open_client_mode()
+        )
+
+        config.add_target_address(
+            snmp_engine,
+            "my-target",
+            udp.DOMAIN_NAME,
+            (self.ip_address, self.port),
+            "my-creds"
+        )
+
+        return snmp_engine
+
+    @classmethod
+    def client(cls) -> str:
+        return "jumpstarter_driver_snmp.client.SNMPServerClient"
+
+    def _snmp_set(self, state: PowerState):
+        result = {"success": False, "error": None}
+
+        def callback(snmpEngine, sendRequestHandle, errorIndication,
+                    errorStatus, errorIndex, varBinds, cbCtx):
+            self.logger.info(f"Callback {errorIndication} {errorStatus} {errorIndex} {varBinds}")
+            if errorIndication:
+                self.logger.info(f"SNMP error: {errorIndication}")
+                result["error"] = f"SNMP error: {errorIndication}"
+            elif errorStatus:
+                self.logger.info(f"SNMP status: {errorStatus}")
+                result["error"] = (
+                    f"SNMP error: {errorStatus.prettyPrint()} at "
+                    f"{varBinds[int(errorIndex) - 1][0] if errorIndex else '?'}"
+                )
+            else:
+                result["success"] = True
+                for oid, val in varBinds:
+                    self.logger.debug(f"{oid.prettyPrint()} = {val.prettyPrint()}")
+            self.logger.info(f"SNMP set result: {result}")
+
+        try:
+            self.logger.info(f"Sending power {state.name} command to {self.host}")
+
+            snmp_engine = self._setup_snmp()
+
+            cmdgen.SetCommandGenerator().send_varbinds(
+                snmp_engine,
+                "my-target",
+                None,
+                "",
+                [(self.full_oid, rfc1902.Integer(state.value))],
+                callback,
+            )
+
+            snmp_engine.open_dispatcher(self.timeout)
+            snmp_engine.close_dispatcher()
+
+            if not result["success"]:
+                raise SNMPError(result["error"])
+
+            return f"Power {state.name} command sent successfully"
+
+        except Exception as e:
+            error_msg = f"SNMP set failed: {str(e)}"
+            self.logger.error(error_msg)
+            raise SNMPError(error_msg) from e
+
+    @export
+    def power_on(self):
```

Comment:
```suggestion
    def on(self):
```

So you can use the default power interface. https://github.com/jumpstarter-dev/jumpstarter/pull/249/files#diff-a6486ddc66753eb801b3d8b638ebe912ae37c36fb9e6d773e5850316a7b77058R42

I need to update my base class in there anyway